src/OpenLoco/src/World/TownManager.cpp

```diff
@@ -29,12 +34,140 @@ namespace OpenLoco::TownManager
         return regs.eax;
     }
 
+    // 0x00497D70
+    static uint32_t copyTownNameToBuffer(const TownNamesObject* namesObj, uint32_t offset, uint16_t index, char* buffer)
+    {
+        auto* offsetPtr = reinterpret_cast<const std::byte*>(namesObj) + offset;
+        auto srcOffset = *reinterpret_cast<const int16_t*>(offsetPtr + index * 2);
+
+        auto* srcPtr = reinterpret_cast<const char*>(offsetPtr + srcOffset);
+        strcpy(buffer, srcPtr);
+
+        uint8_t tail = *reinterpret_cast<const uint8_t*>(srcPtr + strlen(srcPtr) + 1);
+        return tail;
+    }
+
+    // 0x00497A6A
+    static uint8_t townNameFromNamesObject(uint32_t& rand, const char* buffer)
```

Comment:
rand won't be used outside of this function so no need to uint32_t&. All its randomness will be used by this function.